promql/functions.go

```diff
@@ -235,6 +300,48 @@ func funcXincrease(vals []parser.Value, args parser.Expressions, enh *EvalNodeHe
 	return extendedRate(vals, args, enh, true, false)
 }
 
+// Extracts points, rangeStartMsec, rangeEndMsec, rangeSeconds from common params.
+// Note: the range is [rangeStartMsec, rangeEndMsec). That is, every sample in range has the property:
+// rangeStartMsec <= sample.T < rangeEndMsec
+func rangeFromSelectors(vals []parser.Value, args parser.Expressions, enh *EvalNodeHelper) ([]Point, int64, int64, float64) {
```

Comment:
This is wonderful... It makes yIncrease function so much easier to understand, having this extraction of arguments moved outside the function and only passing in what you need.

Reply:
Yeah, that's what it took to get the functions clean and simple! IMO it was the only reasonable way to get the boilerplate out of the functions avoid passing arguments in like `isRate`.

But...I'm wondering about backing away from this refactor when I submit to the Prometheus team, in order to get a more surgical PR that isn't cluttered by refactoring.

The best answer is to submit a prefactor PR first, just like we do here at Invoca. That worked great with the `listen` gem, because Samuel Williams is so reasonable. But I'm not seeing that in the Prometheus team. And TBH, when there's a lot of change going on, refactors of any sort do create a headache because of merge conflicts.